java/src/jmri/jmrit/display/Editor.java

```diff
@@ -1329,6 +1329,40 @@ ActionListener init(Positionable pos, JCheckBoxMenuItem cb) {
         popup.add(hideItem);
     }
 
+    /**
+     * Add a menu entry to set visibility of the Positionable item based on the presence of contents.
+     * If the value is null or empty, the icon is not visible.
+     * This is applicable to memory,  block content and LogixNG global variable labels.
+     *
+     * @param p     the item
+     * @param popup the menu to add the entry to
+     */
+    public void setEmptyHiddenMenu(Positionable p, JPopupMenu popup) {
+        if (p.getDisplayLevel() == BKG) {
+            return;
+        }
+        if (p instanceof BlockContentsIcon || p instanceof MemoryIcon || p instanceof GlobalVariableIcon) {
```

Comment:
This works fine. However, a cleaner way to do this is to add a new interface `SupportsEmptyHidden` with the methods
```
    void setEmptyHidden(boolean enabled);
    boolean isEmptyHidden();
```
The classes BlockContentsIcon, MemoryIcon and GlobalVariableIcon would then implement this interface.

This line would then be:
`if (p instanceof SupportsEmptyHidden) {`

The main benefit with this is that if someone later adds a new Positionable which should support this, the only thing he would need to do is to implement this interface.

Also, if someone else later wants to add something that uses this, like a LogixNG expression `SupportsEmptyHidden`, then it's easy to add that expression without knowing which icons supports it.